pkg/bootstrapper/bootstrapper.go

```diff
@@ -558,7 +566,16 @@ func (wmcb *winNodeBootstrapper) generateInitialKubeletArgs(args map[string]stri
 	if wmcb.nodeIP != "" {
 		kubeletArgs = append(kubeletArgs, "--node-ip="+wmcb.nodeIP)
 	}
-	return kubeletArgs
+
+	hostname, err := cloud.GetKubeletHostnameOverride(wmcb.platformType)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
After seeing this, I am wondering if it would make more sense to introduce a cloud interface
with `GetKubeletHostnameOverride` as method

My original and current thoughts are that it would need to be done only if there were more cloud specific functions to be introduced. Is this the only cloud specific WMCB function required for the CCM work @Fedosin?

I could go either way, I will let others weigh in, if they have an opinion on this.

Reply:
Yeah, nothing else besides that is required. So, I don't think this interface is necessary.